packages/indicators/src/IndicatorApi.ts

```diff
@@ -77,8 +77,8 @@ export class IndicatorApi {
       );
       currentIndicatorCodes = indicatorCodes;
       dataElementCodes.push(...nonIndicatorCodes);
-      const currentAggregations = getAggregationsForBuilders(currentBuilders);
-      aggregations.push(...currentAggregations);
+      const listMapsByBuilder = getAggregationListMapsByBuilder(currentBuilders);
+      aggregationListMap = addEntriesInAggregationListMap(aggregationListMap, listMapsByBuilder);
```

Comment:
Potentially could make better separation of concerns (see my comment about your comment indicating the behaviour of calling code from within the util), and possibly make things easier to understand if:
- instead of running the merge process with each iteration, we just build up an array of either all builders or all listMapsByBuilders
- at the end, we then send that whole array through to the util, which merges and flattens them all (meaning it has the understanding of order etc. within it)

What do you think?

Reply:
Good point, that's definitely doable!